src/Sylius/Behat/Context/Ui/Shop/Checkout/CheckoutShippingContext.php

```diff
@@ -41,21 +42,25 @@ public function __construct(
     }
 
     /**
-     * @Given I have proceeded selecting :shippingMethodName shipping method
-     * @When I proceed with :shippingMethodName shipping method
+     * @Given I have proceeded selecting :shippingMethod shipping method
+     * @When I proceed with :shippingMethod shipping method
      */
-    public function iHaveProceededSelectingShippingMethod($shippingMethodName)
+    public function iHaveProceededSelectingShippingMethod(?ShippingMethodInterface $shippingMethod): void
     {
-        $this->iSelectShippingMethod($shippingMethodName);
+        $this->iSelectShippingMethod($shippingMethod);
         $this->selectShippingPage->nextStep();
     }
 
     /**
-     * @Given I have selected :shippingMethodName shipping method
-     * @When I select :shippingMethodName shipping method
+     * @Given I have selected :shippingMethod shipping method
+     * @When I select :shippingMethod shipping method
      */
-    public function iSelectShippingMethod(string $shippingMethodName): void
+    public function iSelectShippingMethod(?ShippingMethodInterface $shippingMethod): void
     {
+        $shippingMethodName = 'Free';
+        if ($shippingMethod !== null) {
+            $shippingMethodName = $shippingMethod->getName();
+        }
```

Comment:
Why did you add it? Shipping method shouldn't be null

Reply:
Because i reuse function `iSelectShippingMethod` where `ShippingMethod` can be null because function: `iProceedSelectingBillingCountryAndShippingMethod` can have null as a agument